src/Microsoft.AspNetCore.Sockets.Client.Http/ServerSentEventsTransport.cs

```diff
@@ -82,60 +82,65 @@ public Task StartAsync(Uri url, IDuplexPipe application, TransferMode requestedT
 
-                        var consumed = input.Start;
-                        var examined = input.End;
+                    var consumed = input.Start;
+                    var examined = input.End;
 
-                        try
-                        {
-                            var parseResult = _parser.ParseMessage(input, out consumed, out examined, out var buffer);
-
-                            switch (parseResult)
-                            {
-                                case ServerSentEventsMessageParser.ParseResult.Completed:
-                                    await _application.Output.WriteAsync(buffer);
-                                    _parser.Reset();
-                                    break;
-                                case ServerSentEventsMessageParser.ParseResult.Incomplete:
-                                    if (result.IsCompleted)
-                                    {
-                                        throw new FormatException("Incomplete message.");
-                                    }
-                                    break;
-                            }
-                        }
-                        finally
+                    try
+                    {
+                        var parseResult = _parser.ParseMessage(input, out consumed, out examined, out var buffer);
+
+                        switch (parseResult)
                         {
-                            pipelineReader.AdvanceTo(consumed, examined);
+                            case ServerSentEventsMessageParser.ParseResult.Completed:
+                                await _application.Output.WriteAsync(buffer);
+                                _parser.Reset();
+                                break;
+                            case ServerSentEventsMessageParser.ParseResult.Incomplete:
+                                if (result.IsCompleted)
+                                {
+                                    throw new FormatException("Incomplete message.");
+                                }
+                                break;
                         }
                     }
+                    finally
+                    {
+                        pipelineReader.AdvanceTo(consumed, examined);
+                    }
                 }
-                catch (OperationCanceledException)
-                {
-                    _logger.ReceiveCanceled();
-                }
-                finally
+            }
+            catch (OperationCanceledException)
+            {
+                _logger.ReceiveCanceled();
+            }
+            finally
+            {
+                readCancellationRegistration.Dispose();
+                _transportCts.Cancel();
+                try
                 {
-                    readCancellationRegistration.Dispose();
-                    _transportCts.Cancel();
-                    _logger.ReceiveStopped();
+                    stream.Dispose();
```

Comment:
I was going to ask you to ifdef the try catch by meh.

Reply:
ifdef requires cross-compiling